auth/presentation/src/main/java/com/nolawiworkineh/auth/presentation/register/RegisterViewModel.kt

```diff
@@ -3,15 +3,55 @@ package com.nolawiworkineh.auth.presentation.register
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.setValue
+import androidx.compose.runtime.snapshotFlow
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.nolawiworkineh.auth.domain.UserDataValidator
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
 
 class RegisterViewModel(
-
+    private val userDataValidator: UserDataValidator,
 ) : ViewModel() {
 
     var state by mutableStateOf(RegisterState())
         private set
 
+    init {
```

Comment:
There have been some interesting discussions recently related to triggering logic required when a user lands on a particular screen, which I think you may find interesting.

Philipp recently made a video covering this topic. If you prefer reading, I've also linked some great articles about the topic.

[Philipp's video](https://www.youtube.com/watch?v=mNKQ9dc1knI)
https://proandroiddev.com/loading-initial-data-in-launchedeffect-vs-viewmodel-f1747c20ce62
https://proandroiddev.com/loading-initial-data-part-2-clear-all-your-doubts-0f621bfd06a0